### PR TITLE
Add Travis support

### PR DESCRIPTION
**Pull request by gelldur**
Change-Id: I347d4329669d24fb60a2d0e5db6c984f7081f1c6
Signed-off-by: Dawid Drozd <dawid.drozd@arm.com>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  - mkdir -p ~/bob_workspace/src/github.com/google/
+  - mkdir -p ~/bob_workspace/src/github.com/ARM-software/
+  - ln -s "$(pwd)/blueprint/" ~/bob_workspace/src/github.com/google/blueprint
+  - ln -s "$(pwd)"            ~/bob_workspace/src/github.com/ARM-software/bob-build
+
+script:
+  - export GOROOT=$(go env GOROOT)
+  - export BOB_ROOT=$(pwd)
+  - export BOB_WORKSPACE=~/bob_workspace
+  - export GOPATH=${BOB_WORKSPACE}
```
[the remark this answers is not in the document]
Comment (gelldur) [reply .travis.yml]: I also needed set GOPATH to run go test & vet on my local machine in past (go 1.10)

File: .travis/run_all_tests.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+trap "echo '<------------- run_all_tests.sh failed'" ERR
+
+go vet ./...
+RESULT=$?
+if [ $RESULT -ne 0 ]; then
+    echo "Fixes needed, check 'go vet' ❌"
+    echo "########################"
+    go vet ./...
+    echo "########################"
+    exit 1
```
[the remark this answers is not in the document]
Comment (gelldur) [reply .travis/run_all_tests.sh]: I did fail as fast possible but sure I will change

File: .travis/run_all_tests.sh [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+####################
+echo "Check: go vet"
+go vet ./...
+RESULT=$?
+if [ $RESULT -ne 0 ]; then
+    echo "Check 'go vet': FAIL"
+    STATUS_CODE=1
+else
+	echo "Check 'go vet': OK"
```
Comment (chrisdiamand) [inline .travis/run_all_tests.sh]: Should this indentation line up with the 'if' block? (also on lines 24, 37, and 49).

File: core/template_test.go [hunk 4]
```diff
@@ -33,7 +34,14 @@ func setupTestConfig(m map[string]string) *configProperties {
 }
 
 func assertEqual(t *testing.T, a interface{}, b interface{}, msg string) {
-	t.Helper() // To indicate the function is a helper and we're not interested in line numbers coming from it.
+	// Indicate that the function is a helper and we're not
+	// interested in line numbers coming from it. This isn't
+	// present before Go 1.9 (CHECK THE VERSION), so call t.Helper()
```
Comment (chrisdiamand) [inline core/template_test.go]: Please replace "CHECK THE VERSION" with a version which is checked! :)

File: core/toolchain.go [hunk 5]
```diff
@@ -48,7 +48,7 @@ func lookPathSecond(toolUnqualified string, firstHit string) (string, error) {
 			foundFirstHit = true
 		}
 	}
-	return "", &exec.Error{toolUnqualified, exec.ErrNotFound}
+	return "", &exec.Error{Name: toolUnqualified, Err: exec.ErrNotFound}
```
Comment (chrisdiamand) [inline core/toolchain.go]: LGTM!

**Comment by gelldur** [comment]
@chrisdiamand git clone won't fail see `mv ninja ../ninjabin/`

**Comment by chrisdiamand** [comment]
> @chrisdiamand git clone won't fail see `mv ninja ../ninjabin/`

But that's the ninja binary, ~/ninja/ninja -> ~/ninjabin/ninja. The directory containing the source code, ~/ninja, will still exist.

**Comment by gelldur** [comment]
> > @chrisdiamand git clone won't fail see `mv ninja ../ninjabin/`
> 
> But that's the ninja binary, ~/ninja/ninja -> ~/ninjabin/ninja. The directory containing the source code, ~/ninja, will still exist.

Please see travis cache that stores bin output only. Result of git clone will be removed.

**Pull request merged**